src/Swift.Bindings/src/Demangler/DemanglingResults.cs

```diff
@@ -0,0 +1,107 @@
+using BindingsGeneration.Demangling;
+using Xamarin;
+
+/// <summary>
+/// A class to contain results from demangling the set of symbols in a MachO file.
+/// </summary>
+public class DemanglingResults
+{
+    /// <summary>
+    /// Constructs a DemanglingResults object with the desired reductions separated out
+    /// </summary>
+    /// <param name="reductions">An array of reductions</param>
+    DemanglingResults (IReduction [] reductions)
+    {
+        Errors = ArrayOf<ReductionError> (reductions);
```

Comment:
Are the results accessed directly by the parser/emitter, or are they retrieved via the type registrar class?

Reply:
It depends on where it's needed. Probably at the beginning of binding, you would run the demangler in parallel with the code that reads in the .json.abi file.